GoogleForCreators/web-stories-wp
# Animations: Refactor animation state to improve performance

## Context
Original Issue: https://github.com/google/web-stories-wp/issues/9742

Found problem and solution proposed here: https://github.com/google/web-stories-wp/issues/9742#issuecomment-974466215


## Summary
Updates the animation provider and WAAPI animation format to work better with react and media.


## Relevant Technical Choices
- No more composed or changing wrappers from the animation provider makes react not remount the children whenever they change
- No more unnecessary animation regenerations (only generates new animations that have actually updated)
- Increased use-context-selector specificity (coupled with only necessary animation generation) allows for less WAAPIWrapper rerenders (only wrapper that received update rerenders)


## To-do
NA


## User-facing changes
NA


## Testing Instructions





This PR can be tested by following these steps:

1. Click through all background and foreground animations and see that they still work as they did before


(This isn't user facing, but devs can verify this) 
Add a bunch of media with animations to a page and open up the developer tools network tab. Filter network requests by media. See that

 the media isn't reloading when you move an elements position.

**before**
https://user-images.githubusercontent.com/35983235/146572545-cd44e7f0-db03-4fc6-a6ad-8f5f1907f504.mp4

**after**

https://user-images.githubusercontent.com/35983235/146572593-0334fafa-cf49-4cc4-82aa-a1dc80f5f693.mp4




## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9742
